rotkehlchen/db/addressbook.py

```diff
@@ -64,67 +93,106 @@ def add_addressbook_entries(
                     )
                 # Handling both private db (pysqlcipher) and global db (raw sqlite3)
                 except (dbapi2.IntegrityError, sqlite3.IntegrityError) as e:  # pylint: disable=no-member  # noqa: E501
                     raise InputError(
-                        f'Addressbook entry with address "{entry.address}" and name "{entry.name}"'
-                        f' already exists in the address book. Identifier must be unique.',
+                        f'{entry} already exists in the address book. Identifier must be unique.',
                     ) from e
 
     def update_addressbook_entries(
             self,
             book_type: AddressbookType,
             entries: list[AddressbookEntry],
     ) -> None:
+        """Updates names of addressbook entries."""
         with self.write_ctx(book_type) as write_cursor:
             for entry in entries:
-                write_cursor.execute(
-                    'UPDATE address_book SET name = ? WHERE address = ?',
-                    (entry.name, entry.address),
-                )
+                query = 'UPDATE address_book SET name = ? WHERE address = ? AND blockchain = ?'
+                bindings = (entry.name, entry.address, entry.blockchain.value if entry.blockchain else None)  # noqa: E501
+                write_cursor.execute(query, bindings)
                 if write_cursor.rowcount == 0:
                     raise InputError(
-                        f'Addressbook entry with address "{entry.address}" and name "{entry.name}"'
-                        f' doesn\'t exist in the address book. So it cannot be modified.',
+                        f'{entry} doesn\'t exist in the address book. So it cannot be modified.',
                     )
 
     def delete_addressbook_entries(
             self,
             book_type: AddressbookType,
-            addresses: list[ChecksumEvmAddress],
+            chain_addresses: list[OptionalChainAddress],
     ) -> None:
+        """Delete addressbook entries.
+        May raise:
+        - InputError: if any of the addresses that need to be deleted is not present in the
+        addressbook
+        """
+        # fist check that all addresses exist locally. Since each address can have multiple
+        # blockchains entry we need to count them before trying to delete
+        addresses = {chain_address.address for chain_address in chain_addresses}
+        with self.read_ctx(book_type) as read_cursor:
+            read_cursor.execute(
+                f'SELECT DISTINCT address FROM address_book WHERE address IN ({",".join("?"*len(chain_addresses))})',  # noqa: E501
+                list(addresses),
+            )
+            db_addresses = {row[0] for row in read_cursor}
+            if len(addresses) != len(db_addresses):
+                raise InputError(f'Addresses {addresses} are not present in the database')
+
+        delete_without_blockchain = 'DELETE FROM address_book WHERE address = ?'
```

Comment:
Does the API allow both addresses with name + chain and addresses with name + no chain specified in the same call? Cause this seems to complicate the logic here a bit no?

If the API allows this then should it be allowed? How does the frontend use this feature?